### ISSUE TITLE
Imagick Alternative for Image Transformations / Thumbnails

### ISSUE THREAD

**Issue by brusch** [pimcore/pimcore#17080]
### Improvement description

As Imagick PHP extension seems to be abandoned, see: 
https://github.com/Imagick/imagick/issues/640

... we need to look for an alternative. 

Suggestions welcome 😊

**Comment by jdreesen** [comment]
I haven't tried it, but there is https://github.com/libvips/php-vips

**Comment by brusch** [comment]
Yeah, Libvips looks very promising and well maintained👍(and it has an fallback interface to Imagemagick as well).

**Comment by youwe-petervanderwal** [comment]
My hosting colleague suggested http://www.graphicsmagick.org/index.html which is forked from Imagick. However, the PHP PECL extension doesn't seem very well maintained (anymore) https://pecl.php.net/package/gmagick so that might require `exec()` calls to the `gm` binary directly...

**Comment by dpfaffenbauer** [comment]
@youwe-petervanderwal that is exactly the problem with imagemagick, the php extension is not maintained anymore.

**Comment by youwe-petervanderwal** [comment]
> that is exactly the problem with imagemagick, the php extension is not maintained anymore.

I know, hence my "might require `exec` calls" which I know isn't an optimal solution...

> there is https://github.com/libvips/php-vips

Be aware that PHP-VIPS relies on the FFI extension which comes with some warnings https://www.php.net/manual/en/intro.ffi.php . Might be/become a hosting related problem for some users to have this extension enabled.

**Comment by dkarlovi** [comment]
An alternative for simple(r) use cases might be to allow integration with [imgproxy](https://github.com/imgproxy/imgproxy).

This would mean Pimcore would need to know how to generate the correct processing URL, imgproxy does the rest. It might make sense for some scenarios.

**Comment by brusch** [comment]
Libvips looks promising - the fact that it requires FFI is a valid point, but shouldn't be a big issue for most use-cases. 
An integration of Libvips would be anyway optional, so we'd keep GD and Imagick (as long as it makes sense) as an alternative option as well.
